### PR TITLE
Nodejs single account wallet

### LINKED ISSUE #1424: Update NodeJs binding to support "single account wallet"

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Update the nodejs binding to a single account wallet. 

## Links to any relevant issues

Fixes #1424 (the other examples I would update in another PR)

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

`yarn run-example ./how_tos/accounts_and_addresses/create-wallet.ts`

### CODE REVIEW

File: bindings/nodejs/examples/how_tos/accounts_and_addresses/create-wallet.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 // SPDX-License-Identifier: Apache-2.0
 
-import { Wallet, CoinType, initLogger, WalletOptions } from '@iota/sdk';
+import {
+    Wallet,
+    CoinType,
+    initLogger,
+    WalletOptions,
+    SecretManager,
+    Bip44,
```
Comment (thibault-martinez) [inline bindings/nodejs/examples/how_tos/accounts_and_addresses/create-wallet.ts]: unused

Comment (Thoralf-M) [reply bindings/nodejs/examples/how_tos/accounts_and_addresses/create-wallet.ts]: It was used 3 minutes ago :cry:

Comment (Thoralf-M) [reply bindings/nodejs/examples/how_tos/accounts_and_addresses/create-wallet.ts]: Removed

File: bindings/nodejs/examples/.env.example
Comment (thibault-martinez) [inline bindings/nodejs/examples/.env.example:1]: Why ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

File: bindings/nodejs/examples/.env.example
Comment (Thoralf-M) [reply bindings/nodejs/examples/.env.example:1]: Deleted by accident

File: bindings/core/src/method_handler/secret_manager.rs [hunk 2]
```diff
@@ -124,6 +124,13 @@ where
             if let Some(secret_manager) = secret_manager.downcast::<StrongholdSecretManager>() {
                 secret_manager.store_mnemonic(mnemonic).await?;
                 Response::Ok
+            } else if let Some(secret_manager) = secret_manager.downcast::<SecretManager>() {
```
Comment (thibault-martinez) [inline bindings/core/src/method_handler/secret_manager.rs:127]: Was there still an issue with this? I think I saw a discussion somewhere

File: bindings/nodejs/lib/types/wallet/wallet.ts [hunk 3]
```diff
@@ -1,14 +1,155 @@
...
+    /** Whether to sync Basic outputs. */
+    basicOutputs?: boolean;
+    /** Whether to sync Account outputs. */
+    accountOutputs?: boolean;
+    /** Whether to sync NFT outputs. */
+    nftOutputs?: boolean;
+}
+
+/** Specifies what outputs should be synced for the address of an account output. */
+export interface AliasSyncOptions {
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/wallet/wallet.ts]: This has been renamed in the Rust PR, want to do it as well?

File: bindings/nodejs/lib/types/wallet/prepared-transaction.ts [hunk 4]
```diff
@@ -11,20 +11,20 @@ import {
...
  */
 export class PreparedTransaction {
     readonly _preparedData: PreparedTransactionData;
-    readonly _account: Account;
+    readonly _wallet: Wallet;
 
     /**
      * @param preparedData Prepared data to sign and submit a transaction.
-     * @param account A wallet account.
+     * @param wallet A wallet wallet.
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/wallet/prepared-transaction.ts]: wallet wallet :trollface:

File: bindings/nodejs/lib/wallet/wallet.ts [hunk 5]
```diff
@@ -155,27 +137,6 @@ export class Wallet {
         return this.methodHandler.getSecretManager();
     }
 
-    /**
-     * Generate an address without storing it.
-     */
-    async generateEd25519Address(
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/wallet/wallet.ts:161]: Shouldn't we hold off removing this?

File: bindings/core/src/method_handler/secret_manager.rs (hunk 2 above)
Comment (Thoralf-M) [reply bindings/core/src/method_handler/secret_manager.rs:127]: There was only an issue without this being added

File: bindings/nodejs/lib/wallet/wallet.ts (hunk 5 above)
Comment (Thoralf-M) [reply bindings/nodejs/lib/wallet/wallet.ts:161]: That's a different generate address, this one here would just generate an address and return it, so exactly the same as the secret manager
The discussion is about the generate address inside of the wallet if you don't provide an address

File: bindings/nodejs/lib/types/wallet/wallet.ts (hunk 3 above)
Comment (Thoralf-M) [reply bindings/nodejs/lib/types/wallet/wallet.ts]: Done

File: bindings/nodejs/lib/types/wallet/prepared-transaction.ts (hunk 4 above)
Comment (Thoralf-M) [reply bindings/nodejs/lib/types/wallet/prepared-transaction.ts]: Thanks, fixed

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**